src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double STEP_UPHILL = 1.5d;
+    private Object[] dataCollection;
```

Comment:
You should not use an Object, you should use a parameterized type T, fix everywhere
```suggestion
    private T[] dataCollection;
```